pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -122,57 +125,121 @@ func (r *ReconcileNode) createOrUpdateCephCrash(node corev1.Node, tolerations []
 	return controllerutil.CreateOrUpdate(context.TODO(), r.client, deploy, mutateFunc)
 }
 
-// createOrUpdateCephCron is a wrapper around controllerutil.CreateOrUpdate
-func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, cephVersion *version.CephVersion) (controllerutil.OperationResult, error) {
-	cronJob := &v1beta1.CronJob{
+// createV1CronJob creates and return v1.CronJob for k8s version atleast than '1.21.0'
+func createV1CronJob(namespace string) *v1.CronJob {
+	return &v1.CronJob{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      prunerName,
-			Namespace: cephCluster.GetNamespace(),
+			Namespace: namespace,
 		},
 	}
-	err := controllerutil.SetControllerReference(&cephCluster, cronJob, r.scheme)
-	if err != nil {
-		return controllerutil.OperationResultNone, errors.Errorf("failed to set owner reference of deployment %q", cronJob.Name)
+}
+
+// createV1Beta1CronJob creates and return v1beta.CronJob for k8s version lesser than '1.21.0'
```

Comment:
I would suggest renaming it to `getV1Beta1CronJob` since this method is not creating the actual cronJob
```suggestion
// getV1Beta1CronJob returns v1beta1.CronJob for k8s version lower than '1.21.0'
```